src/IECoreMaya/FromMayaArrayDataConverter.cpp

```diff
@@ -38,11 +38,33 @@
 
 #include "IECore/VectorTypedData.h"
 
+#include "IECore/GeometricTypedData.h"
+
 using namespace IECore;
 
 namespace IECoreMaya
 {
 
+namespace Detail
```

Comment:
I didn't on purpose because it potentially changes the behavior of the code where MVectorArray is used as a non-vector float3 data for some reason, but it is a natural conversion and may be safe since there should not be a lot of redundant code that transforms a "non-vector" vector value. Do you think it's ok to add a specialization?